Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1001,6 +999,10 @@ void GridTrackSizingAlgorithm::cacheBaselineAlignedItem(const RenderBox& item, G
     if (GridLayoutFunctions::isOrthogonalParent(*m_renderGrid, *item.parent()))
         axis = axis == GridColumnAxis ? GridRowAxis : GridColumnAxis;
 
+    const auto* gridItemParent = dynamicDowncast<RenderGrid>(item.parent());
+    if (gridItemParent && axis == GridColumnAxis && !m_renderGrid->isSubgridRows() && gridItemParent->isSubgridOf(GridLayoutFunctions::flowAwareDirectionForChild(*m_renderGrid, item, GridTrackSizingDirection::ForRows), *m_renderGrid))
```

Comment:
I'm finding this condition quite hard to parse and understand.

Is this something where we could pass down a flag through the recursion to say 'we're inside a subgrid of the root, in the axis that the root calls Rows'?

Failing that, maybe just a comment trying to explain what we're catching here.

Reply:
I went ahead and started to keep track of a bool to represent when we want to cache row subgrids and is initially set to true when called on the root grid. We can check to see if that bool is true and if the item's parent is a row subgrid. 

It makes the logic slightly shorter and hopefully easier to read and understand. Let me know what you think!